Arduino IDE Not compiling

Hi all! 
Recently decided to change from the Arduino library inbuilt and directly download from GitHub as I want to take advantage of the logger as having some issues. But compiling my code just gives tons of errors. Tried compiling for Seeduinio XAIO, Teensy & Generica ESP module. Also loaded example code and no luck there either.  Any help would be greatly appreciated! 
Thingsboard 0.6.0 compiles perfectly and runs. Github SDK doesn't.

ArduinoJSON - 6.19.4
MQTT PubSub Client - 0.1.3
Arduino HTTP Client - 0.4.0


`In file included from /var/folders/3h/k85zg0y13zn0fh1mp3fnpwvc0000gp/T/arduino_modified_sketch_35148/0000-arduino_send_telemetry.ino:8:0:
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:269:28: error: 'function' in namespace 'std' does not name a template type
     using processFn = std::function<RPC_Response(const RPC_Data &data)>;
                            ^~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:277:49: error: 'processFn' has not been declared
     inline RPC_Callback(const char* methodName, processFn cb)
                                                 ^~~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:282:5: error: 'processFn' does not name a type
     processFn   m_cb;       // Callback to call
     ^~~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h: In constructor 'RPC_Callback::RPC_Callback()':
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:273:19: error: class 'RPC_Callback' does not have any field named 'm_cb'
       : m_name(), m_cb(nullptr) {  }
                   ^~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h: In constructor 'RPC_Callback::RPC_Callback(const char*, int)':
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:278:29: error: class 'RPC_Callback' does not have any field named 'm_cb'
       : m_name(methodName), m_cb(cb) {  }
                             ^~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h: At global scope:
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:292:34: error: ISO C++ forbids declaration of 'type name' with no type [-fpermissive]
     using processFn = const std::function<void(const Shared_Attribute_Data &data)>;
                                  ^~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:292:29: error: expected ';'
     using processFn = const std::function<void(const Shared_Attribute_Data &data)>;
                             ^~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:301:101: error: 'processFn' has not been declared
     inline Shared_Attribute_Callback(const InputIterator& first_itr, const InputIterator& last_itr, processFn cb)
                                                                                                     ^~~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:306:48: error: expected ')' before 'cb'
     inline Shared_Attribute_Callback(processFn cb)
                                                ^~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:311:5: error: 'processFn' does not name a type
     processFn                      m_cb;    // Callback to call
     ^~~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h: In constructor 'Shared_Attribute_Callback::Shared_Attribute_Callback()':
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:296:18: error: class 'Shared_Attribute_Callback' does not have any field named 'm_cb'
       : m_att(), m_cb(nullptr) {  }
                  ^~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h: In constructor 'Shared_Attribute_Callback::Shared_Attribute_Callback(const InputIterator&, const InputIterator&, int)':
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:302:37: error: class 'Shared_Attribute_Callback' does not have any field named 'm_cb'
       : m_att(first_itr, last_itr), m_cb(cb) {  }
                                     ^~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h: At global scope:
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:321:28: error: 'function' in namespace 'std' does not name a template type
     using processFn = std::function<void(const Shared_Attribute_Data &data)>;
                            ^~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:328:56: error: expected ')' before 'cb'
     inline Shared_Attribute_Request_Callback(processFn cb)
                                                        ^~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:333:5: error: 'processFn' does not name a type
     processFn       m_cb;           // Callback to call
     ^~~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h: In constructor 'Shared_Attribute_Request_Callback::Shared_Attribute_Request_Callback()':
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:325:27: error: class 'Shared_Attribute_Request_Callback' does not have any field named 'm_cb'
       : m_request_id(0U), m_cb(nullptr) {  }
                           ^~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h: In member function 'void ThingsBoardSized<PayloadSize, MaxFieldsAmt, Logger>::setClient(arduino::Client&)':
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:410:33: error: 'bind' is not a member of 'std'
       m_client.setCallback(std::bind(&ThingsBoardSized::on_message, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3));
                                 ^~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:410:33: note: suggested alternative: 'find'
       m_client.setCallback(std::bind(&ThingsBoardSized::on_message, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3));
                                 ^~~~
                                 find
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:410:80: error: 'std::placeholders' has not been declared
       m_client.setCallback(std::bind(&ThingsBoardSized::on_message, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3));
                                                                                ^~~~~~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:410:103: error: 'std::placeholders' has not been declared
       m_client.setCallback(std::bind(&ThingsBoardSized::on_message, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3));
                                                                                                       ^~~~~~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:410:126: error: 'std::placeholders' has not been declared
       m_client.setCallback(std::bind(&ThingsBoardSized::on_message, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3));
                                                                                                                              ^~~~~~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h: In member function 'const uint8_t ThingsBoardSized<PayloadSize, MaxFieldsAmt, Logger>::detect_size(const char*, ...)':
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:979:7: error: there are no arguments to 'va_start' that depend on a template parameter, so a declaration of 'va_start' must be available [-fpermissive]
       va_start(args, msg);
       ^~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:979:7: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
In file included from /Users/andy_shepshep/Documents/Arduino/libraries/ArduinoJson/src/ArduinoJson/Variant/VariantData.hpp:7:0,
                 from /Users/andy_shepshep/Documents/Arduino/libraries/ArduinoJson/src/ArduinoJson/Variant/SlotFunctions.hpp:8,
                 from /Users/andy_shepshep/Documents/Arduino/libraries/ArduinoJson/src/ArduinoJson/Array/ArrayIterator.hpp:7,
                 from /Users/andy_shepshep/Documents/Arduino/libraries/ArduinoJson/src/ArduinoJson/Array/ArrayRef.hpp:8,
                 from /Users/andy_shepshep/Documents/Arduino/libraries/ArduinoJson/src/ArduinoJson.hpp:24,
                 from /Users/andy_shepshep/Documents/Arduino/libraries/ArduinoJson/src/ArduinoJson.h:9,
                 from /Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:21,
                 from /var/folders/3h/k85zg0y13zn0fh1mp3fnpwvc0000gp/T/arduino_modified_sketch_35148/0000-arduino_send_telemetry.ino:8:
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:983:47: error: there are no arguments to 'vsnprintf_P' that depend on a template parameter, so a declaration of 'vsnprintf_P' must be available [-fpermissive]
       const int32_t result = JSON_STRING_SIZE(vsnprintf_P(nullptr, 0U, msg, args));
                                               ^
In file included from /var/folders/3h/k85zg0y13zn0fh1mp3fnpwvc0000gp/T/arduino_modified_sketch_35148/0000-arduino_send_telemetry.ino:8:0:
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:984:7: error: there are no arguments to 'assert' that depend on a template parameter, so a declaration of 'assert' must be available [-fpermissive]
       assert(result >= 0);
       ^~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:985:7: error: there are no arguments to 'va_end' that depend on a template parameter, so a declaration of 'va_end' must be available [-fpermissive]
       va_end(args);
       ^~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h: In member function 'void ThingsBoardSized<PayloadSize, MaxFieldsAmt, Logger>::process_rpc_message(char*, uint8_t*, uint32_t)':
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:1058:24: error: 'const class RPC_Callback' has no member named 'm_cb'
           if (callback.m_cb == nullptr || callback.m_name == nullptr) {
                        ^~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:1086:26: error: 'const class RPC_Callback' has no member named 'm_cb'
             r = callback.m_cb(param);
                          ^~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:1092:26: error: 'const class RPC_Callback' has no member named 'm_cb'
             r = callback.m_cb(param);
                          ^~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h: In member function 'void ThingsBoardSized<PayloadSize, MaxFieldsAmt, Logger>::process_shared_attribute_update_message(char*, uint8_t*, uint32_t)':
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:1218:52: error: '__gnu_cxx::__alloc_traits<std::allocator<Shared_Attribute_Callback> >::value_type {aka class Shared_Attribute_Callback}' has no member named 'm_cb'
         if (m_sharedAttributeUpdateCallbacks.at(i).m_cb == nullptr) {
                                                    ^~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:1225:50: error: '__gnu_cxx::__alloc_traits<std::allocator<Shared_Attribute_Callback> >::value_type {aka class Shared_Attribute_Callback}' has no member named 'm_cb'
           m_sharedAttributeUpdateCallbacks.at(i).m_cb(data);
                                                  ^~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:1260:48: error: '__gnu_cxx::__alloc_traits<std::allocator<Shared_Attribute_Callback> >::value_type {aka class Shared_Attribute_Callback}' has no member named 'm_cb'
         m_sharedAttributeUpdateCallbacks.at(i).m_cb(data);
                                                ^~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h: In member function 'void ThingsBoardSized<PayloadSize, MaxFieldsAmt, Logger>::process_shared_attribute_request_message(char*, uint8_t*, uint32_t)':
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:1292:53: error: '__gnu_cxx::__alloc_traits<std::allocator<Shared_Attribute_Request_Callback> >::value_type {aka class Shared_Attribute_Request_Callback}' has no member named 'm_cb'
         if (m_sharedAttributeRequestCallbacks.at(i).m_cb == nullptr) {
                                                     ^~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:1305:49: error: '__gnu_cxx::__alloc_traits<std::allocator<Shared_Attribute_Request_Callback> >::value_type {aka class Shared_Attribute_Request_Callback}' has no member named 'm_cb'
         m_sharedAttributeRequestCallbacks.at(i).m_cb(data);
                                                 ^~~~
                ^~~~~~~~~
/var/folders/3h/k85zg0y13zn0fh1mp3fnpwvc0000gp/T/arduino_modified_sketch_35148/0000-arduino_send_telemetry.ino:20:16: note: suggested alternative: 'HttpClient'
 ThingsBoard tb(espClient);
                ^~~~~~~~~
                HttpClient
In file included from /var/folders/3h/k85zg0y13zn0fh1mp3fnpwvc0000gp/T/arduino_modified_sketch_35148/0000-arduino_send_telemetry.ino:8:0:
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h: In instantiation of 'const bool ThingsBoardSized<PayloadSize, MaxFieldsAmt, Logger>::sendKeyval(const char*, T, bool) [with T = int; unsigned int PayloadSize = 64; unsigned int MaxFieldsAmt = 8; Logger = ThingsBoardDefaultLogger]':
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:505:24:   required from 'const bool ThingsBoardSized<PayloadSize, MaxFieldsAmt, Logger>::sendTelemetryInt(const char*, int) [with unsigned int PayloadSize = 64; unsigned int MaxFieldsAmt = 8; Logger = ThingsBoardDefaultLogger]'
/var/folders/3h/k85zg0y13zn0fh1mp3fnpwvc0000gp/T/arduino_modified_sketch_35148/0000-arduino_send_telemetry.ino:44:40:   required from here
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:1032:43: error: call of overloaded 'sendTelemetryJson(ArduinoJson::JsonVariant&)' is ambiguous
       return telemetry ? sendTelemetryJson(object) : sendAttributeJSON(object);
                          ~~~~~~~~~~~~~~~~~^~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:529:23: note: candidate: const bool ThingsBoardSized<PayloadSize, MaxFieldsAmt, Logger>::sendTelemetryJson(const char*) [with unsigned int PayloadSize = 64; unsigned int MaxFieldsAmt = 8; Logger = ThingsBoardDefaultLogger]
     inline const bool sendTelemetryJson(const char *json) {
                       ^~~~~~~~~~~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:545:23: note: candidate: const bool ThingsBoardSized<PayloadSize, MaxFieldsAmt, Logger>::sendTelemetryJson(const JsonObject&) [with unsigned int PayloadSize = 64; unsigned int MaxFieldsAmt = 8; Logger = ThingsBoardDefaultLogger; ArduinoJson::JsonObject = ArduinoJson6194_F1::ObjectRef]
     inline const bool sendTelemetryJson(const JsonObject& jsonObject) {
                       ^~~~~~~~~~~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:1032:71: error: call of overloaded 'sendAttributeJSON(ArduinoJson::JsonVariant&)' is ambiguous
       return telemetry ? sendTelemetryJson(object) : sendAttributeJSON(object);
                                                      ~~~~~~~~~~~~~~~~~^~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:594:23: note: candidate: const bool ThingsBoardSized<PayloadSize, MaxFieldsAmt, Logger>::sendAttributeJSON(const char*) [with unsigned int PayloadSize = 64; unsigned int MaxFieldsAmt = 8; Logger = ThingsBoardDefaultLogger]
     inline const bool sendAttributeJSON(const char *json) {
                       ^~~~~~~~~~~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:610:23: note: candidate: const bool ThingsBoardSized<PayloadSize, MaxFieldsAmt, Logger>::sendAttributeJSON(const JsonObject&) [with unsigned int PayloadSize = 64; unsigned int MaxFieldsAmt = 8; Logger = ThingsBoardDefaultLogger; ArduinoJson::JsonObject = ArduinoJson6194_F1::ObjectRef]
     inline const bool sendAttributeJSON(const JsonObject& jsonObject) {
                       ^~~~~~~~~~~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h: In instantiation of 'const bool ThingsBoardSized<PayloadSize, MaxFieldsAmt, Logger>::sendKeyval(const char*, T, bool) [with T = float; unsigned int PayloadSize = 64; unsigned int MaxFieldsAmt = 8; Logger = ThingsBoardDefaultLogger]':
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:515:24:   required from 'const bool ThingsBoardSized<PayloadSize, MaxFieldsAmt, Logger>::sendTelemetryFloat(const char*, float) [with unsigned int PayloadSize = 64; unsigned int MaxFieldsAmt = 8; Logger = ThingsBoardDefaultLogger]'
/var/folders/3h/k85zg0y13zn0fh1mp3fnpwvc0000gp/T/arduino_modified_sketch_35148/0000-arduino_send_telemetry.ino:45:41:   required from here
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:1032:43: error: call of overloaded 'sendTelemetryJson(ArduinoJson::JsonVariant&)' is ambiguous
       return telemetry ? sendTelemetryJson(object) : sendAttributeJSON(object);
                          ~~~~~~~~~~~~~~~~~^~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:529:23: note: candidate: const bool ThingsBoardSized<PayloadSize, MaxFieldsAmt, Logger>::sendTelemetryJson(const char*) [with unsigned int PayloadSize = 64; unsigned int MaxFieldsAmt = 8; Logger = ThingsBoardDefaultLogger]
     inline const bool sendTelemetryJson(const char *json) {
                       ^~~~~~~~~~~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:545:23: note: candidate: const bool ThingsBoardSized<PayloadSize, MaxFieldsAmt, Logger>::sendTelemetryJson(const JsonObject&) [with unsigned int PayloadSize = 64; unsigned int MaxFieldsAmt = 8; Logger = ThingsBoardDefaultLogger; ArduinoJson::JsonObject = ArduinoJson6194_F1::ObjectRef]
     inline const bool sendTelemetryJson(const JsonObject& jsonObject) {
                       ^~~~~~~~~~~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:1032:71: error: call of overloaded 'sendAttributeJSON(ArduinoJson::JsonVariant&)' is ambiguous
       return telemetry ? sendTelemetryJson(object) : sendAttributeJSON(object);
                                                      ~~~~~~~~~~~~~~~~~^~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:594:23: note: candidate: const bool ThingsBoardSized<PayloadSize, MaxFieldsAmt, Logger>::sendAttributeJSON(const char*) [with unsigned int PayloadSize = 64; unsigned int MaxFieldsAmt = 8; Logger = ThingsBoardDefaultLogger]
     inline const bool sendAttributeJSON(const char *json) {
                       ^~~~~~~~~~~~~~~~~
/Users/andy_shepshep/Documents/Arduino/libraries/thingsboard-arduino-sdk-master/src/ThingsBoard.h:610:23: note: candidate: const bool ThingsBoardSized<PayloadSize, MaxFieldsAmt, Logger>::sendAttributeJSON(const JsonObject&) [with unsigned int PayloadSize = 64; unsigned int MaxFieldsAmt = 8; Logger = ThingsBoardDefaultLogger; ArduinoJson::JsonObject = ArduinoJson6194_F1::ObjectRef]
     inline const bool sendAttributeJSON(const JsonObject& jsonObject) {
                       ^~~~~~~~~~~~~~~~~`

--- a/examples/0002-arduino_rpc/0002-arduino_rpc.ino
+++ b/examples/0002-arduino_rpc/0002-arduino_rpc.ino
@@ -122,7 +122,7 @@ void loop() {
     // Perform a subscription. All consequent data processing will happen in
     // processTemperatureChange() and processSwitchChange() functions,
     // as denoted by callbacks vector.
-    if (!tb.RPC_Subscribe(callbacks)) {
+    if (!tb.RPC_Subscribe(callbacks.cbegin(), callbacks.cend())) {
       Serial.println("Failed to subscribe for RPC");
       return;
     }

--- a/library.json
+++ b/library.json
@@ -6,7 +6,7 @@
         "type": "git",
         "url": "https://github.com/thingsboard/ThingsBoard-Arduino-MQTT-SDK"
     },
-    "version": "0.6",
+    "version": "0.7",
     "examples": "examples/*/*.ino",
     "frameworks": "arduino"
 }

--- a/library.properties
+++ b/library.properties
@@ -1,5 +1,5 @@
 name=ThingsBoard
-version=0.6
+version=0.7
 author=ThingsBoard Team
 maintainer=ThingsBoard Team
 sentence=ThingsBoard library for Arduino.

--- a/src/ThingsBoard.h
+++ b/src/ThingsBoard.h
@@ -103,6 +103,7 @@ constexpr char ATT_KEY_NOT_FOUND[] = PSTR("Shared attribute key not found");
 constexpr char ATT_REQUEST_CB_IS_NULL[] = PSTR("Shared attribute request callback is NULL");
 constexpr char CALLING_REQUEST_ATT_CB[] = PSTR("Calling subscribed callback for response id (%u)");
 constexpr char TOO_MANY_JSON_FIELDS[] = PSTR("Too many JSON fields passed (%u), increase MaxFieldsAmt (%u) accordingly");
+constexpr char TOO_BIG_JSON_TEXT[] = PSTR("Too big of a JSON passed (%u), increase PayloadSize (%u) accordingly");
 constexpr char CB_ON_MESSAGE[] = PSTR("Callback on_message from topic: (%s)");
 constexpr char CONNECT_FAILED[] = PSTR("Connecting to server failed");
 
@@ -542,7 +543,7 @@ class ThingsBoardSized
     }
 
     // Sends custom JSON telemetry JsonObject to the ThingsBoard.
-    inline const bool sendTelemetryJson(const JsonObject& jsonObject) {
+    inline const bool sendTelemetryJson(const JsonObject& jsonObject, const uint32_t& jsonSize) {
       const uint32_t json_object_size = jsonObject.size();
       if (MaxFieldsAmt < json_object_size) {
         char message[detect_size(TOO_MANY_JSON_FIELDS, json_object_size, MaxFieldsAmt)];
@@ -550,9 +551,14 @@ class ThingsBoardSized
         Logger::log(message);
         return false;
       }
-      const uint32_t json_size = JSON_STRING_SIZE(measureJson(jsonObject));
-      char json[json_size];
-      serializeJson(jsonObject, json, json_size);
+      else if (PayloadSize < jsonSize) {
+        char message[detect_size(TOO_BIG_JSON_TEXT, jsonSize, PayloadSize)];
+        snprintf_P(message, sizeof(message), TOO_BIG_JSON_TEXT, jsonSize, PayloadSize);
+        Logger::log(message);
+        return false;
+      }
+      char json[jsonSize];
+      serializeJson(jsonObject, json, jsonSize);
       return sendTelemetryJson(json);
     }
 
@@ -607,7 +613,7 @@ class ThingsBoardSized
     }
 
     // Sends custom JsonObject with attributes to the ThingsBoard.
-    inline const bool sendAttributeJSON(const JsonObject& jsonObject) {
+    inline const bool sendAttributeJSON(const JsonObject& jsonObject, const uint32_t& jsonSize) {
       const uint32_t json_object_size = jsonObject.size();
       if (MaxFieldsAmt < json_object_size) {
         char message[detect_size(TOO_MANY_JSON_FIELDS, json_object_size, MaxFieldsAmt)];
@@ -615,9 +621,14 @@ class ThingsBoardSized
         Logger::log(message);
         return false;
       }
-      const uint32_t json_size = JSON_STRING_SIZE(measureJson(jsonObject));
-      char json[json_size];
-      serializeJson(jsonObject, json, json_size);
+      else if (PayloadSize < jsonSize) {
+        char message[detect_size(TOO_BIG_JSON_TEXT, jsonSize, PayloadSize)];
+        snprintf_P(message, sizeof(message), TOO_BIG_JSON_TEXT, jsonSize, PayloadSize);
+        Logger::log(message);
+        return false;
+      }
+      char json[jsonSize];
+      serializeJson(jsonObject, json, jsonSize);
       return sendAttributeJSON(json);
     }
 
@@ -703,7 +714,7 @@ class ThingsBoardSized
 
       currentFirmwareInfoObject[CURR_FW_TITLE_KEY] = currFwTitle;
       currentFirmwareInfoObject[CURR_FW_VER_KEY] = currFwVersion;
-      return sendTelemetryJson(currentFirmwareInfoObject);
+      return sendTelemetryJson(currentFirmwareInfoObject, JSON_STRING_SIZE(measureJson(currentFirmwareInfoObject)));
     }
 
     inline const bool Firmware_Send_State(const char* currFwState) {
@@ -712,7 +723,7 @@ class ThingsBoardSized
       JsonObject currentFirmwareStateObject = currentFirmwareState.to<JsonObject>();
 
       currentFirmwareStateObject[CURR_FW_STATE_KEY] = currFwState;
-      return sendTelemetryJson(currentFirmwareStateObject);
+      return sendTelemetryJson(currentFirmwareStateObject, JSON_STRING_SIZE(measureJson(currentFirmwareStateObject)));
     }
 
     inline const bool Firmware_OTA_Subscribe() {

--- a/src/ThingsBoard.h
+++ b/src/ThingsBoard.h
@@ -12,16 +12,25 @@
 #include <ArduinoHttpClient.h>
 #endif
 
-#if defined(ESP8266)
-#include <Updater.h>
-#elif defined(ESP32)
-#include <Update.h>
-#endif
-
 #include <PubSubClient.h>
 #include <ArduinoJson.h>
 #include <vector>
 #include <array>
+#include <functional>
+#include <pgmspace.h>
+
+#if defined(ESP8266)
+#include <Updater.h>
+// Add defines used that or not included in the pgm header.
+#ifndef snprintf_P
+#define snprintf_P    snprintf
+#endif // snprintf_P
+#ifndef vsnprintf_P
+#define vsnprintf_P   vsnprintf
+#endif // vsnprintf_P
+#elif defined(ESP32)
+#include <Update.h>
+#endif
 
 // Local includes.
 #include "ThingsBoardDefaultLogger.h"

--- a/src/ThingsBoard.h
+++ b/src/ThingsBoard.h
@@ -103,7 +103,6 @@ constexpr char ATT_KEY_NOT_FOUND[] = PSTR("Shared attribute key not found");
 constexpr char ATT_REQUEST_CB_IS_NULL[] = PSTR("Shared attribute request callback is NULL");
 constexpr char CALLING_REQUEST_ATT_CB[] = PSTR("Calling subscribed callback for response id (%u)");
 constexpr char TOO_MANY_JSON_FIELDS[] = PSTR("Too many JSON fields passed (%u), increase MaxFieldsAmt (%u) accordingly");
-constexpr char TOO_BIG_JSON_TEXT[] = PSTR("Too big of a JSON passed (%u), increase PayloadSize (%u) accordingly");
 constexpr char CB_ON_MESSAGE[] = PSTR("Callback on_message from topic: (%s)");
 constexpr char CONNECT_FAILED[] = PSTR("Connecting to server failed");
 
@@ -533,7 +532,7 @@ class ThingsBoardSized
       }
 
       const uint32_t json_size = JSON_STRING_SIZE(strlen(json));
-      if (json_size > PayloadSize) {
+      if (PayloadSize < json_size) {
         char message[detect_size(INVALID_BUFFER_SIZE, PayloadSize, json_size)];
         snprintf_P(message, sizeof(message), INVALID_BUFFER_SIZE, PayloadSize, json_size);
         Logger::log(message);
@@ -544,16 +543,10 @@ class ThingsBoardSized
 
     // Sends custom JSON telemetry JsonObject to the ThingsBoard.
     inline const bool sendTelemetryJson(const JsonObject& jsonObject, const uint32_t& jsonSize) {
-      const uint32_t json_object_size = jsonObject.size();
-      if (MaxFieldsAmt < json_object_size) {
-        char message[detect_size(TOO_MANY_JSON_FIELDS, json_object_size, MaxFieldsAmt)];
-        snprintf_P(message, sizeof(message), TOO_MANY_JSON_FIELDS, json_object_size, MaxFieldsAmt);
-        Logger::log(message);
-        return false;
-      }
-      else if (PayloadSize < jsonSize) {
-        char message[detect_size(TOO_BIG_JSON_TEXT, jsonSize, PayloadSize)];
-        snprintf_P(message, sizeof(message), TOO_BIG_JSON_TEXT, jsonSize, PayloadSize);
+      const uint32_t jsonObjectSize = jsonObject.size();
+      if (MaxFieldsAmt < jsonObjectSize) {
+        char message[detect_size(TOO_MANY_JSON_FIELDS, jsonObjectSize, MaxFieldsAmt)];
+        snprintf_P(message, sizeof(message), TOO_MANY_JSON_FIELDS, jsonObjectSize, MaxFieldsAmt);
         Logger::log(message);
         return false;
       }
@@ -603,7 +596,7 @@ class ThingsBoardSized
       }
 
       const uint32_t json_size = JSON_STRING_SIZE(strlen(json));
-      if (json_size > PayloadSize) {
+      if (PayloadSize < json_size) {
         char message[detect_size(INVALID_BUFFER_SIZE, PayloadSize, json_size)];
         snprintf_P(message, sizeof(message), INVALID_BUFFER_SIZE, PayloadSize, json_size);
         Logger::log(message);
@@ -614,16 +607,10 @@ class ThingsBoardSized
 
     // Sends custom JsonObject with attributes to the ThingsBoard.
     inline const bool sendAttributeJSON(const JsonObject& jsonObject, const uint32_t& jsonSize) {
-      const uint32_t json_object_size = jsonObject.size();
-      if (MaxFieldsAmt < json_object_size) {
-        char message[detect_size(TOO_MANY_JSON_FIELDS, json_object_size, MaxFieldsAmt)];
-        snprintf_P(message, sizeof(message), TOO_MANY_JSON_FIELDS, json_object_size, MaxFieldsAmt);
-        Logger::log(message);
-        return false;
-      }
-      else if (PayloadSize < jsonSize) {
-        char message[detect_size(TOO_BIG_JSON_TEXT, jsonSize, PayloadSize)];
-        snprintf_P(message, sizeof(message), TOO_BIG_JSON_TEXT, jsonSize, PayloadSize);
+      const uint32_t jsonObjectSize = jsonObject.size();
+      if (MaxFieldsAmt < jsonObjectSize) {
+        char message[detect_size(TOO_MANY_JSON_FIELDS, jsonObjectSize, MaxFieldsAmt)];
+        snprintf_P(message, sizeof(message), TOO_MANY_JSON_FIELDS, jsonObjectSize, MaxFieldsAmt);
         Logger::log(message);
         return false;
       }
@@ -1116,7 +1103,7 @@ class ThingsBoardSized
       }
 
       const uint32_t json_size = JSON_STRING_SIZE(measureJson(respBuffer));
-      if (json_size > PayloadSize) {
+      if (PayloadSize < json_size) {
         char message[detect_size(INVALID_BUFFER_SIZE, PayloadSize, json_size)];
         snprintf_P(message, sizeof(message), INVALID_BUFFER_SIZE, PayloadSize, json_size);
         Logger::log(message);

--- a/src/ThingsBoard.h
+++ b/src/ThingsBoard.h
@@ -384,6 +384,9 @@ class ThingsBoardSized
       : m_client()
       , m_requestId(0)
       , m_qos(enableQoS)
+#if defined(ESP8266) || defined(ESP32)
+      , m_fwResponseCallback(std::bind(&ThingsBoardSized::Firmware_Shared_Attribute_Received, this, std::placeholders::_1))
+#endif
     {
       reserve_callback_size();
       setClient(client);
@@ -394,6 +397,9 @@ class ThingsBoardSized
       : m_client()
       , m_requestId(0)
       , m_qos(false)
+#if defined(ESP8266) || defined(ESP32)
+      , m_fwResponseCallback(std::bind(&ThingsBoardSized::Firmware_Shared_Attribute_Received, this, std::placeholders::_1))
+#endif
     {
       reserve_callback_size();
     }
@@ -728,8 +734,7 @@ class ThingsBoardSized
 
       // Request the firmware informations
       const std::array<const char*, 5U> fwSharedKeys {FW_CHKS_KEY, FW_CHKS_ALGO_KEY, FW_SIZE_KEY, FW_TITLE_KEY, FW_VER_KEY};
-      Shared_Attribute_Request_Callback callback(std::bind(&ThingsBoardSized::Firmware_Shared_Attribute_Received, this, std::placeholders::_1));
-      return Shared_Attributes_Request(fwSharedKeys.cbegin(), fwSharedKeys.cend(), callback);
+      return Shared_Attributes_Request(fwSharedKeys.cbegin(), fwSharedKeys.cend(), m_fwResponseCallback);
     }
 
     inline const bool Firmware_Send_FW_Info(const char* currFwTitle, const char* currFwVersion) {
@@ -1414,6 +1419,7 @@ class ThingsBoardSized
     uint16_t m_fwSize;
     const char* m_fwChecksum;
     std::function<void(const bool&)> m_fwUpdatedCallback;
+    Shared_Attribute_Request_Callback m_fwResponseCallback;
     uint16_t m_fwChunkReceive;
 #endif
 

--- a/src/ThingsBoard.h
+++ b/src/ThingsBoard.h
@@ -567,6 +567,20 @@ class ThingsBoardSized
       return sendTelemetryJson(json);
     }
 
+    // Sends custom JSON telemetry JsonVariant to the ThingsBoard.
+    inline const bool sendTelemetryJson(const JsonVariant& jsonVariant, const uint32_t& jsonSize) {
+      const uint32_t jsonVariantSize = jsonVariant.size();
+      if (MaxFieldsAmt < jsonVariantSize) {
+        char message[detect_size(TOO_MANY_JSON_FIELDS, jsonVariantSize, MaxFieldsAmt)];
+        snprintf_P(message, sizeof(message), TOO_MANY_JSON_FIELDS, jsonVariantSize, MaxFieldsAmt);
+        Logger::log(message);
+        return false;
+      }
+      char json[jsonSize];
+      serializeJson(jsonVariant, json, jsonSize);
+      return sendTelemetryJson(json);
+    }
+
     //----------------------------------------------------------------------------
     // Attribute API
 
@@ -629,6 +643,20 @@ class ThingsBoardSized
       char json[jsonSize];
       serializeJson(jsonObject, json, jsonSize);
       return sendAttributeJSON(json);
+    }
+
+    // Sends custom JsonVariant with attributes to the ThingsBoard.
+    inline const bool sendAttributeJSON(const JsonVariant& jsonVariant, const uint32_t& jsonSize) {
+      const uint32_t jsonVariantSize = jsonVariant.size();
+      if (MaxFieldsAmt < jsonVariantSize) {
+        char message[detect_size(TOO_MANY_JSON_FIELDS, jsonVariantSize, MaxFieldsAmt)];
+        snprintf_P(message, sizeof(message), TOO_MANY_JSON_FIELDS, jsonVariantSize, MaxFieldsAmt);
+        Logger::log(message);
+        return false;
+      }
+      char json[jsonSize];
+      serializeJson(jsonVariant, json, jsonSize);
+      return sendTelemetryJson(json);
     }
 
     //----------------------------------------------------------------------------

--- a/src/ThingsBoard.h
+++ b/src/ThingsBoard.h
@@ -7,6 +7,7 @@
 #ifndef ThingsBoard_h
 #define ThingsBoard_h
 
+// Library includes.
 #if !defined(ESP8266) && !defined(ESP32)
 #include <ArduinoHttpClient.h>
 #endif
@@ -20,6 +21,7 @@
 #include <PubSubClient.h>
 #include <ArduinoJson.h>
 #include <vector>
+#include <array>
 
 // Local includes.
 #include "ThingsBoardDefaultLogger.h"
@@ -682,17 +684,15 @@ class ThingsBoardSized
         return false;
       }
 
-      // Request the firmware informations
-      const std::array<const char*, 5U> fwSharedKeys {FW_CHKS_KEY, FW_CHKS_ALGO_KEY, FW_SIZE_KEY, FW_TITLE_KEY, FW_VER_KEY};
-      if (!Shared_Attributes_Request(fwSharedKeys.cbegin(), fwSharedKeys.cend(), Shared_Attribute_Request_Callback(std::bind(&ThingsBoardSized::Firmware_Shared_Attribute_Received, this, std::placeholders::_1)))) {
-        return false;
-      }
-
       // Set private members needed for update.
       m_currFwTitle = currFwTitle;
       m_currFwVersion = currFwVersion;
       m_fwUpdatedCallback = updatedCallback;
-      return true;
+
+      // Request the firmware informations
+      const std::array<const char*, 5U> fwSharedKeys {FW_CHKS_KEY, FW_CHKS_ALGO_KEY, FW_SIZE_KEY, FW_TITLE_KEY, FW_VER_KEY};
+      Shared_Attribute_Request_Callback callback(std::bind(&ThingsBoardSized::Firmware_Shared_Attribute_Received, this, std::placeholders::_1));
+      return Shared_Attributes_Request(fwSharedKeys.cbegin(), fwSharedKeys.cend(), callback);
     }
 
     inline const bool Firmware_Send_FW_Info(const char* currFwTitle, const char* currFwVersion) {
@@ -1037,7 +1037,7 @@ class ThingsBoardSized
         return false;
       }
 
-      return telemetry ? sendTelemetryJson(object, JSON_STRING_SIZE(measureJson(object)) : sendAttributeJSON(object, JSON_STRING_SIZE(measureJson(object));
+      return telemetry ? sendTelemetryJson(object, JSON_STRING_SIZE(measureJson(object))) : sendAttributeJSON(object, JSON_STRING_SIZE(measureJson(object)));
     }
 
     // Processes RPC message
@@ -1355,7 +1355,7 @@ class ThingsBoardSized
         }
       }
 
-      return telemetry ? sendTelemetryJson(object, JSON_STRING_SIZE(measureJson(object)) : sendAttributeJSON(object, JSON_STRING_SIZE(measureJson(object));
+      return telemetry ? sendTelemetryJson(object, JSON_STRING_SIZE(measureJson(object))) : sendAttributeJSON(object, JSON_STRING_SIZE(measureJson(object)));
     }
 
     PubSubClient m_client; // PubSub MQTT client instance.
@@ -1376,7 +1376,7 @@ class ThingsBoardSized
     const char* m_fwState;
     uint16_t m_fwSize;
     const char* m_fwChecksum;
-    const std::function<void(const bool&)> m_fwUpdatedCallback;
+    std::function<void(const bool&)> m_fwUpdatedCallback;
     uint16_t m_fwChunkReceive;
 #endif
 
@@ -1552,7 +1552,7 @@ class ThingsBoardHttpSized
         }
       }
 
-      return telemetry ? sendTelemetryJson(object, JSON_STRING_SIZE(measureJson(object)) : sendAttributeJSON(object, JSON_STRING_SIZE(measureJson(object));
+      return telemetry ? sendTelemetryJson(object, JSON_STRING_SIZE(measureJson(object))) : sendAttributeJSON(object, JSON_STRING_SIZE(measureJson(object)));
     }
 
     // Sends single key-value in a generic way.
@@ -1567,7 +1567,7 @@ class ThingsBoardHttpSized
         return false;
       }
 
-      return telemetry ? sendTelemetryJson(object, JSON_STRING_SIZE(measureJson(object)) : sendAttributeJSON(object, JSON_STRING_SIZE(measureJson(object));
+      return telemetry ? sendTelemetryJson(object, JSON_STRING_SIZE(measureJson(object))) : sendAttributeJSON(object, JSON_STRING_SIZE(measureJson(object)));
     }
 
     HttpClient m_client;

--- a/src/ThingsBoard.h
+++ b/src/ThingsBoard.h
@@ -159,6 +159,7 @@ constexpr char FW_STATE_CHKS_ERROR[] = PSTR("CHECKSUM ERROR");
 
 // Log messages.
 constexpr char NO_FW[] = PSTR("No new firmware assigned on the given device");
+constexpr char EMPTY_FW[] = PSTR("Given firmware was NULL");
 constexpr char FW_UP_TO_DATE[] = PSTR("Firmware is already up to date");
 constexpr char FW_NOT_FOR_US[] = PSTR("Firmware is not for us (title is different)");
 constexpr char FW_CHKS_ALGO_NOT_SUPPORTED[] = PSTR("Checksum algorithm is not supported, please use MD5 only");
@@ -730,6 +731,12 @@ class ThingsBoardSized
     }
 
     inline void Firmware_Shared_Attribute_Received(const Shared_Attribute_Data& data) {
+      // Print out firmware shared attributes.
+      int jsonSize = JSON_STRING_SIZE(measureJson(data));
+      char buffer[jsonSize];
+      serializeJson(data, buffer, jsonSize);
+      Logger::log(buffer);
+
       // Check if firmware is available for our device
       if (!data.containsKey(FW_VER_KEY) || !data.containsKey(FW_TITLE_KEY)) {
         Logger::log(NO_FW);
@@ -743,21 +750,24 @@ class ThingsBoardSized
       const char* fw_checksum_algorithm = data[FW_CHKS_ALGO_KEY].as<const char*>();
       m_fwSize = data[FW_SIZE_KEY].as<const uint16_t>();
 
+      if (fw_title == nullptr || fw_version == nullptr || m_currFwTitle == nullptr || m_currFwVersion == nullptr || m_fwChecksum == nullptr || fw_checksum_algorithm == nullptr) {
+        Logger::log(EMPTY_FW);
+        Firmware_Send_State(FW_STATE_NO_FW);
+        return;
+      }
       // If firmware is the same, we do not update it
-      if (strncmp_P(m_currFwTitle, fw_title, strlen(m_currFwTitle)) == 0 && strncmp_P(m_currFwVersion, fw_version, strlen(m_currFwVersion) == 0)) {
+      else if (strncmp_P(m_currFwTitle, fw_title, JSON_STRING_SIZE(strlen(m_currFwTitle))) == 0 && strncmp_P(m_currFwVersion, fw_version, JSON_STRING_SIZE(strlen(m_currFwVersion))) == 0) {
         Logger::log(FW_UP_TO_DATE);
         Firmware_Send_State(FW_STATE_UP_TO_DATE);
         return;
       }
-
       // If firmware title is not the same, we quit now
-      if (strncmp_P(m_currFwTitle, fw_title, strlen(m_currFwTitle)) != 0) {
+      else if (strncmp_P(m_currFwTitle, fw_title, JSON_STRING_SIZE(strlen(m_currFwTitle))) != 0) {
         Logger::log(FW_NOT_FOR_US);
         Firmware_Send_State(FW_STATE_NO_FW);
         return;
       }
-
-      if (strncmp_P(FW_CHECKSUM_VALUE, fw_checksum_algorithm, strlen(FW_CHECKSUM_VALUE)) != 0) {
+      else if (strncmp_P(FW_CHECKSUM_VALUE, fw_checksum_algorithm, JSON_STRING_SIZE(strlen(FW_CHECKSUM_VALUE))) != 0) {
         Logger::log(FW_CHKS_ALGO_NOT_SUPPORTED);
         Firmware_Send_State(FW_STATE_INVALID_CHKS);
         return;
@@ -1027,7 +1037,7 @@ class ThingsBoardSized
         return false;
       }
 
-      return telemetry ? sendTelemetryJson(object) : sendAttributeJSON(object);
+      return telemetry ? sendTelemetryJson(object, JSON_STRING_SIZE(measureJson(object)) : sendAttributeJSON(object, JSON_STRING_SIZE(measureJson(object));
     }
 
     // Processes RPC message
@@ -1345,7 +1355,7 @@ class ThingsBoardSized
         }
       }
 
-      return telemetry ? sendTelemetryJson(object) : sendAttributeJSON(object);
+      return telemetry ? sendTelemetryJson(object, JSON_STRING_SIZE(measureJson(object)) : sendAttributeJSON(object, JSON_STRING_SIZE(measureJson(object));
     }
 
     PubSubClient m_client; // PubSub MQTT client instance.
@@ -1542,7 +1552,7 @@ class ThingsBoardHttpSized
         }
       }
 
-      return telemetry ? sendTelemetryJson(object) : sendAttributeJSON(object);
+      return telemetry ? sendTelemetryJson(object, JSON_STRING_SIZE(measureJson(object)) : sendAttributeJSON(object, JSON_STRING_SIZE(measureJson(object));
     }
 
     // Sends single key-value in a generic way.
@@ -1557,7 +1567,7 @@ class ThingsBoardHttpSized
         return false;
       }
 
-      return telemetry ? sendTelemetryJson(object) : sendAttributeJSON(object);
+      return telemetry ? sendTelemetryJson(object, JSON_STRING_SIZE(measureJson(object)) : sendAttributeJSON(object, JSON_STRING_SIZE(measureJson(object));
     }
 
     HttpClient m_client;

